Framework/Geometry/src/Instrument/ComponentInfo.cpp

```diff
@@ -239,6 +241,167 @@ double ComponentInfo::solidAngle(const size_t componentIndex,
+ * @param mutableBB : Output bounding box. This will be grown.
+ * @param mutableDetExclusions : Output detector exclusions to append to. These
+ * are ranges of detector indices that we do NOT need to consider for future
+ * bounding box calculations for detectors.
+ * @param mutableIterator : Iterator to the next non-detector component.
+ */
+template <typename IteratorT>
+void ComponentInfo::growBoundingBoxAsBankOfTubes(
+    size_t index, const BoundingBox *reference, BoundingBox &mutableBB,
+    std::stack<std::pair<size_t, size_t>> &mutableDetExclusions,
+    IteratorT &mutableIterator) const {
+  auto innerRangeComp = m_componentInfo->componentRangeInSubtree(index);
+  auto nSubComponents = innerRangeComp.end() - innerRangeComp.begin() - 1;
+  auto innerRangeDet = m_componentInfo->detectorRangeInSubtree(index);
+  auto nSubDetectors =
+      std::distance(innerRangeDet.begin(), innerRangeDet.end());
+  auto nY = nSubDetectors / nSubComponents;
+  size_t bottomIndex = *innerRangeDet.begin();
+  size_t lastIndex = *(innerRangeDet.end() - 1);
+  while (bottomIndex < lastIndex) {
+    auto topIndex = bottomIndex + (nY - 1);
+    mutableBB.grow(componentBoundingBox(bottomIndex, reference));
+    mutableBB.grow(componentBoundingBox(topIndex, reference));
+    bottomIndex += nY;
+  }
+  mutableDetExclusions.emplace(
+      std::make_pair(*innerRangeDet.begin(), *(innerRangeDet.end() - 1)));
+  mutableIterator = innerRangeComp.rend();
+}
+
+/**
+ * Grow the bounding box on the basis that the component described by index is a
+ * tube containing only detectors. Only the tube tips need to be treated.
+ *
+ * Find the start and end index
+ * Grow the bounding box by the top and bottom of the tube.
+ * Record detectors that form part of this bank that are enclosed and therefore
+ * already inclusive of the external bounding box
+ *
+ * @param index : Index of the component to get the bounding box for
+ * @param reference : Reference bounding box (optional)
+ * @param mutableBB : Output bounding box. This will be grown.
+ * @param mutableDetExclusions : Output detector exclusions to append to. These
+ * are ranges of detector indices that we do NOT need to consider for future
+ * bounding box calculations for detectors.
+ * @param mutableIterator : Iterator to the next non-detector component.
+ */
+template <typename IteratorT>
+void ComponentInfo::growBoundingBoxAsTube(
+    size_t index, const BoundingBox *reference, BoundingBox &mutableBB,
+    std::stack<std::pair<size_t, size_t>> &mutableDetExclusions,
+    IteratorT &mutableIterator) const {
+
+  auto rangeDet = m_componentInfo->detectorRangeInSubtree(index);
+  if (!rangeDet.empty()) {
+    auto startIndex = *rangeDet.begin();
+    auto endIndex = *(rangeDet.end() - 1);
+    mutableBB.grow(componentBoundingBox(startIndex, reference));
+    mutableBB.grow(componentBoundingBox(endIndex, reference));
+    mutableDetExclusions.emplace(std::make_pair(startIndex, endIndex));
```

Comment:
I am a bit confused (also in the sibling methods, apart from the `byDetectors` variant): We are creating exclusions, but not handling any. Where is the handling happening?

Reply:
They are passed into `growBoundingBoxByDetectors`. I've put that into it's own method, but the workings are essentially unchanged from before.